File: 0053-maximum-subarray/0053-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int maxSum = Integer.MIN_VALUE;
        int currentSum = 0;
        
        for (int i : nums) {
            currentSum += i;
            
            if (currentSum > maxSum) {
                maxSum = currentSum;
            }
            if (currentSum < 0) {
                currentSum = 0;
            }
        }
        
        return maxSum;
    }
}

File: 0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int cur_sum = 0;
        int max_sum = INT_MIN;
        int n = nums.size();
        for (int i = 0; i < n; i++) {
            cur_sum += nums[i];
            max_sum = max(max_sum, cur_sum);
            if (cur_sum < 0) {
                cur_sum = 0;
            }
        }
        return max_sum;
    }
};